plugins/module_utils/foreman_helper.py

```diff
@@ -834,10 +834,10 @@ def auto_lookup_nested_entities(self):
             if entity_spec.get('type') in {'nested_list'}:
                 for nested_key, nested_spec in self.foreman_spec[key]['foreman_spec'].items():
                     for item in self.foreman_params.get(key, []):
-                        if (nested_key in item and nested_spec.get('resolve', True) and not nested_spec.get('resolved', False)
+                        if (nested_key in item and nested_spec.get('resolve', True)
+                                and item[nested_key] is not None and not isinstance(item[nested_key], (dict, list))
```

Comment:
@mdellweg I wonder if we should call this `is_resolved(…)` and use it in the normal tracking too?

Reply:
So you say: "Never save the `resolved` attribute, and always use the heuristic"?
Sounds good to me.

Reply:
Yeah, that was the idea. Let's merge this as is and then refactor afterwards.